docs/user-guide/source-hydrator.md

```diff
@@ -171,6 +171,52 @@ Argo CD will only push changes to the `hydrateTo` branch, it will not create a P
 changes to the `syncSource` branch. You will need to use your own tooling to move the changes from the `hydrateTo` 
 branch to the `syncSource` branch.
 
+## Hydrating to a Different Repository
+
+The source hydrator supports hydrating manifests to a separate repository, not just a different branch in the same repository. This allows you to keep your dry source repository clean and push rendered manifests to dedicated GitOps repositories.
+
+To hydrate to a different repository, set the `repoURL` field in the `hydrateTo` configuration:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: my-app
+spec:
+  project: my-project
+  destination:
+    server: https://kubernetes.default.svc
+    namespace: default
+  sourceHydrator:
+    drySource:
+      repoURL: https://github.com/org/app-configs
+      path: apps/my-app
+      targetRevision: main
+    syncSource:
+      repoURL: https://github.com/org/gitops-manifests
+      targetBranch: main
+      path: clusters/prod/my-app
+    hydrateTo:
+      repoURL: https://github.com/org/gitops-manifests
+      targetBranch: staging
+      path: clusters/prod/my-app
+```
+
+In this example:
+- The dry source is in `app-configs` repository
+- The hydrated manifests are pushed to the `staging` branch of the `gitops-manifests` repository
+- Argo CD syncs from the `main` branch of the `gitops-manifests` repository
```

Comment:
```suggestion
- Argo CD syncs from the `main` branch of the `gitops-manifests` repository
```
can you explain this part clearly in the docs?

Reply:
I think based on the outcome of https://github.com/argoproj/argo-cd/pull/25464/files#r2578968496 this section in general needs a rewrite.

If we decide to move along with the current implementation, I would update the docs here to cover both cases (drySource+syncSource=repoURL,drySource+syncSource+hydrateTo=repoURL) in separate sections?

If we decide to move along with only the syncSource field, I would update this to only cover the syncSource field (without hydrateTo in the example application)?